### ISSUE TITLE
wow.profile.get_account_profile_summary returns 403 even if hardcoded to use access_token provided by the Blizzard API website for testing

### ISSUE THREAD

**Issue by Frendin** [lostcol0ny/blizzardapi2#7]
Hey there.

I've had this working for a while and then it suddenly broke and now I'm now entirely sure if this is a valid bug or on my end.

Here's the snippet of my code that I am using first to get my profile info. I do some more sorting with the returned JSON later but it's irrelevant to the bug at the moment.
```
def ScrapeAPICharacterFaction(region, locale, realm, character):
    clientID = <MY CLIENT ID HERE>
    clientSecret = <MY CLIENT PASSWORD HERE>
    #clientAccessToken = CreateAccessToken(clientID, clientSecret, region)
    clientAccessToken = <ACCESS TOKEN GENERATED FROM THE BLIZZARD API WEBSITE>
    realm = realm.capitalize()
    character = character.capitalize()
    apiClient = BlizzardApi(clientID, clientSecret)
    #factionJSON = apiClient.wow.profile.get_account_profile_summary(str(region), str(locale), clientAccessToken["access_token"])
    factionJSON = apiClient.wow.profile.get_account_profile_summary(str(region), str(locale), str(clientAccessToken))

print(ScrapeAPICharacterFaction("us", "en_US", "sargeras", "frendin"))
```

`requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://us.api.blizzard.com/profile/user/wow?namespace=profile-us&locale=en_US`

Basically right now I have my Client ID and Client Secret hardcoded for testing purposes and managed to get an access_token function working properly yesterday, until I started encountering 403 errors this morning. Once that happened, I resorted back to going to the Blizzard API website, inputting my ID and Secret and manually hardcoding in the access_token it gives me for further testing but even that is returning a 403 error. I did regress backwards from Python 3.12 to Python 3.11 yesterday but it was still working before I stopped for the day. 

Again not sure if this is user error or not but wanted to call it out either way.

Thanks for maintaining this library. It's a total godsend.

**Comment by lostcol0ny** [comment]
I see the issue. You're using the account profile API endpoint, which is protected and requires an OAuth2 access token to use ([which is a whole different can of worms](https://develop.battle.net/documentation/guides/using-oauth/authorization-code-flow)). If what you're trying to do is get your character's faction, try `get_character_profile_summary` instead:

`
from blizzardapi2 import BlizzardApi

clientID = <MY CLIENT ID HERE>
clientSecret = <MY CLIENT PASSWORD HERE>

apiClient = BlizzardApi(clientID, clientSecret)

region = "us"
locale = "en_US"

def get_character_profile(realm, character):
    profile_info = apiClient.wow.profile.get_character_profile_summary(region, locale, realm, character)
    return profile_info

character_faction = get_character_profile("sargeras", "frendin")["faction"]["name"])

print(character_faction)
`

That will return a json blob describing the character you used as input, indexed to the name of your faction.

Also, the library handles getting your client credentials for you, so you don't need to add that step in your own code. Just provide your ID and secret and the rest happens behind the scenes.

Take a look at the bottom of the readme. You'll see the following API endpoints are protected, and you can read up more about how to access them programmatically. It's tricky and I'm not entirely convinced this API library implements it correctly (part of me being a hobbyist Python coder and not a real developer).

    GET /oauth/userinfo*
    GET /profile/user/wow
    GET /profile/user/wow/protected-character/{realm-id}-{character-id}
    GET /profile/user/wow/collections
    GET /profile/user/wow/collections/pets
    GET /profile/user/wow/collections/mounts

**Comment by Frendin** [comment]
Hey hey! Thanks for the response,

Fellow hobbyist here. I thought I had to do the OAuth (which was working before, I omitted it from the snippet) in order to find my character's faction but if I can simply get it without having to do so, that would be a godsend. Thanks so much for this. I'll give this a shot today and respond back with how it went.

**Comment by Frendin** [comment]
Okay just gave that a shot and it's totally working. I feel so dumb for having missed that. Thank you for pointing it out! 

I still don't know if something changed on the back end since my original issue is still there, but it's nice I can completely work around it by using a much simpler API call. Thank you so much. I'll leave this bug open, but feel free to close it if everything looks fine for you.

**Comment by lostcol0ny** [comment]
If I had a dollar for every time I felt dumb working with this API, I could have paid someone to write the library for me.

I believe that using the protected API URLs is something most apps do separately from something like this library. With Raider.io, for example, you can log in using Battle.net and that grants them access to see all your account info. That's what those protected endpoints do. It gives an application access to more than just public character info and lets them see your Battle.net ID, all your characters, etc. There's a specific workflow you need for that and I don't know how to do it without setting up a URL for the callback. There are solutions, I just don't know how to use 'em.

Anyway, hopefully the rest of the library can give you access to the info you're looking for. I'll close this issue out, but feel free to post another one if you come across another problem. You can use the [Discussions](https://github.com/lostcol0ny/blizzardapi2/discussions) tab at the top to talk about the library. That's there for general conversations about it, where Issues is more meant for things not working as intended (like this).
